functions.cmake

```diff
@@ -228,13 +233,31 @@ function(_add_library)
       add_library(${ARGV} ${VERSION_FILE_PATH})
     else()
       add_library(${ARGV})
+      list(GET ARGV 0 CLR_CMAKE_CURRENT_TARGET)
+      set_target_properties(
+        ${CLR_CMAKE_CURRENT_TARGET} 
+        PROPERTIES
+        VS_GLOBAL_CL_MPCount ${CLR_CMAKE_CPU_CORES})
     endif(NOT WIN32)
     list(FIND CLR_CROSS_COMPONENTS_LIST ${ARGV0} INDEX)  
     if (DEFINED CLR_CROSS_COMPONENTS_LIST AND ${INDEX} EQUAL -1)  
      set_target_properties(${ARGV0} PROPERTIES EXCLUDE_FROM_ALL 1)  
     endif()  
 endfunction()
 
+function(_add_custom_target)
```

Comment:
Why does the custom target need this option? Custom targets don't build anything themselves, they are just kind of proxies to related custom_command that actually does some work.

Reply:
I need to inject <CL_MPCount>x</CL_MPCount> property into vcxproj which actually builds code. It just happens that mechanism to perform that injection with VS_GLOBAL_<variable> feature works only on targets which translate directly to vcxproj on Windows. Custom targets do that but custom commands not - it is just CMake implementation detail which has surfaced in public API.

Reply:
Ok, that would make sense, but as for the single use of this new _add_custom_target that you have added, the mscordaccore_exp doesn't result in CL invocation, it invokes lib.exe instead. Does the option still apply?

Reply:
I agree that centralizing the change is a good thing. So then it seems we should change all add_custom_target usages by the new _add_custom_target. There are 16 usages of that.

Reply:
I will do that then